File: foodify_api/src/models/Restaurant.model.ts
import mongoose from "mongoose";

const restaurantSchema = new mongoose.Schema({
    name: { type: String, required: true },
    image: { type: String, required: true },
    description: { type: String, required: true },
    rating: { type: Number, default: 4.5 },
    deliveryTime: { type: String, required: true },
    deliveryFee: { type: Number, required: true },
    categories: [{ type: String }],
    isOpen: { type: Boolean, default: true },
    address: { type: String, required: true },
    phone: { type: String, required: true },
}, { timestamps: true });

export const RestaurantModel = mongoose.model("Restaurant", restaurantSchema);

File: foodify_mobile/lib/features/home/data/models/restaurant_model.dart
import '../../domain/entities/restaurant_entity.dart';

class RestaurantModel extends RestaurantEntity {
  const RestaurantModel({
    required super.id,
    required super.name,
    required super.image,
    required super.description,
    required super.rating,
    required super.deliveryTime,
    required super.deliveryFee,
    required super.categories,
    required super.isOpen,
    required super.address,
    required super.phone,
  });

  factory RestaurantModel.fromJson(Map<String, dynamic> json) {
    return RestaurantModel(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      image: json['image'] ?? '',
      description: json['description'] ?? '',
      rating: (json['rating'] ?? 0).toDouble(),
      deliveryTime: json['deliveryTime'] ?? '',
      deliveryFee: (json['deliveryFee'] ?? 0).toDouble(),
      categories: List<String>.from(json['categories'] ?? []),
      isOpen: json['isOpen'] ?? true,
      address: json['address'] ?? '',
      phone: json['phone'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'image': image,
      'description': description,
      'rating': rating,
      'deliveryTime': deliveryTime,
      'deliveryFee': deliveryFee,
      'categories': categories,
      'isOpen': isOpen,
      'address': address,
      'phone': phone,
    };
  }
}
